src/Clock.elm

```diff
@@ -0,0 +1,103 @@
+module Clock exposing (Model, Msg(..), init, main, subscriptions, update, view)
+
+import Browser
+import Html exposing (..)
+import Html.Attributes exposing (style)
+import Task
+import Time
+
+
+
+-- MAIN
+
+
+main =
+    Browser.element
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+
+-- MODEL
+
+
+type alias Model =
+    { zone : Time.Zone
+    , time : Time.Posix
+    }
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( Model Time.utc (Time.millisToPosix 0)
+    , Task.perform AdjustTimeZone Time.here
+    )
+
+
+
+-- UPDATE
+
+
+type Msg
+    = Tick Time.Posix
+    | AdjustTimeZone Time.Zone
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        Tick newTime ->
+            ( { model | time = newTime }
+            , Cmd.none
+            )
+
+        AdjustTimeZone newZone ->
+            ( { model | zone = newZone }
+            , Cmd.none
+            )
+
+
+
+-- SUBSCRIPTIONS
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Time.every 100 Tick
+
+
+
+-- VIEW
+
+
+view : Model -> Html Msg
+view model =
+    let
+        hour =
+            String.fromInt (Time.toHour model.zone model.time)
+
+        minute =
+            String.fromInt (Time.toMinute model.zone model.time)
+
+        second =
+            String.fromInt (Time.toSecond model.zone model.time)
```

Comment:
There's some duplication here - the pattern `String.fromInt (_ model.zone model.time)` repeats 3 times.
It can be extracted to a helper function:
```elm
let
    extractTime : (Time.Zone -> Time.Posix -> Int) -> String
    extractTime toTimeUnit =
        String.fromInt <| toTimeUnit model.zone model.time

    hour : String
    hour =
        extractTime Time.toHour

    minute : String
    minute =
        extractTime Time.toMinute

    second : String
    second =
        extractTime Time.toSecond

    rgb : String
    rgb =
        "rgb(" ++ (String.join "," [hour, minute, second]) ++ ")"
in
...
```

Reply:
If you look carefully, you notice we still have duplication - `extractTime` is used similarly 3 times. This means we can do better:
```elm
let
    extractTime : (Time.Zone -> Time.Posix -> Int) -> String
    extractTime toTimeUnit =
        String.fromInt <| toTimeUnit model.zone model.time

    rgb : String
    rgb =
        "rgb(" ++ (String.join "," <| List.map extractTime [hour, minute, second]) ++ ")"
in
...
```